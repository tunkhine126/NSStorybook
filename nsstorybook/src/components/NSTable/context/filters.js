import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

export const NSFiltersContext = createContext();

export const NSFiltersProvider = ({
  children,
  filterCategories,
  lastUpdateProps,
  dateRangeProps,
  handleSelected,
}) => {
  const types = {
    multi: {},
    levels: {},
    single: {},
    date: {},
  };

  const { lastUpdate } = lastUpdateProps;

  ['multi', 'levels', 'single', 'date'].forEach(type => {
    filterCategories
      .filter(c => c.type === type)
      .forEach(({ id }) => {
        types[type][id] = [];
      });
  });

  const [singleSelectOptions, setSingleSelectOptions] = useState(types.single);
  const [multiSelectOptions, setMultiSelectOptions] = useState(types.multi);
  const [levelSelectOptions, setLevelSelectOptions] = useState(types.levels);
  const [showCalendar, setShowCalendar] = useState(
    lastUpdate.query === 'custom'
  );
  const [localDateRange, setlocalDateRange] = useState({
    before: null,
    after: null,
  });

  return (
    <NSFiltersContext.Provider
      value={{
        multiSelect: { multiSelectOptions, setMultiSelectOptions },
        levelSelect: { levelSelectOptions, setLevelSelectOptions },
        singleSelect: { singleSelectOptions, setSingleSelectOptions },
        calendarProps: { showCalendar, setShowCalendar },
        localDateProps: { localDateRange, setlocalDateRange },
        lastUpdateProps,
        dateRangeProps,
        filterCategories,
        handleSelected,
      }}
    >
      {children}
    </NSFiltersContext.Provider>
  );
};

NSFiltersProvider.propTypes = {
  children: PropTypes.any.isRequired,
  dateRangeProps: PropTypes.object.isRequired,
  lastUpdateProps: PropTypes.object.isRequired,
  filterCategories: PropTypes.array.isRequired,
  handleSelected: PropTypes.func.isRequired,
};
