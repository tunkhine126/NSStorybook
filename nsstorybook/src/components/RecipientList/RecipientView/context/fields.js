import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const FieldsContext = createContext();

export const FieldsProvider = ({ children, ...rest }) => {
  const [activePhotoField, setActivePhotoField] = useState(null);
  const [recipientPhoto, setRecipientPhoto] = useState({});

  return (
    <FieldsContext.Provider
      value={{
        activePhotoProps: { activePhotoField, setActivePhotoField },
        recipientPhotoProps: { recipientPhoto, setRecipientPhoto },
        rest,
      }}
    >
      {children}
    </FieldsContext.Provider>
  );
};

FieldsProvider.propTypes = {
  children: PropTypes.any.isRequired,
};
