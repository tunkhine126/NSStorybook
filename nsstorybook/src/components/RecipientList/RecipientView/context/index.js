import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const RecipientViewContext = createContext();

export const RecipientViewProvider = ({
  children,
  tabChange,
  rdData,
  data,
  profileDetails,
  primaryRecipient,
  match,
  ...rest
}) => {
  const [toaster, setToaster] = useState({
    show: false,
    recipientUuid: null,
    field: null,
    undoHandler: null,
  });
  const [saveState, setSaveState] = useState({});

  return (
    <RecipientViewContext.Provider
      value={{
        data,
        rdData,
        tabChange,
        profileDetails,
        primaryRecipient,
        match,
        toasterProps: { toaster, setToaster },
        saveProps: { saveState, setSaveState },
        rest,
      }}
    >
      {children}
    </RecipientViewContext.Provider>
  );
};

RecipientViewProvider.propTypes = {
  children: PropTypes.any.isRequired,
  tabChange: PropTypes.func.isRequired,
  profileDetails: PropTypes.object.isRequired,
  primaryRecipient: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  rdData: PropTypes.any.isRequired,
  match: PropTypes.any.isRequired,
};
