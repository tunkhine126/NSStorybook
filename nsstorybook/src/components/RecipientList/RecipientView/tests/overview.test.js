/**
 *
 * Tests for OverView
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import OverView from '../overview';
import { RecipientViewProvider } from '../context';

describe('<OverView />', () => {
  const props = {
    profileDetails: {
      recipientName: 'Country',
      childRecipients: 'Communities',
    },
    data: {
      name: 'Test',
      avatarUrl: null,
      internalId: '3erf23-r323d-gewf3-233',
      children: [],
    },
    match: {
      isExact: true,
      params: { id: '07c538ec-f5c8-11e9-a1ad-22000be1c01d' },
      path: '/families/family-list/family/:id',
      url: '/families/family-list/family/07c538ec-f5c8-11e9-a1ad-22000be1c01d',
    },
    rdData: {},
    tabChange: () => {},
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient={false}>
          <OverView value={50} />
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient>
          <OverView value={50} />
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
