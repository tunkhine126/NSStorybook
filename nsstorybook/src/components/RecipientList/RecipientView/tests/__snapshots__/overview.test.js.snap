// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverView /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-container-277 MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between"
>
  <div
    class="MuiGrid-root makeStyles-column1-287 MuiGrid-item"
  >
    <div
      class="MuiPaper-root makeStyles-paper-315 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-widgetHeader-326 makeStyles-widgetPadding-327 MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Communities
      </p>
      <hr
        class="MuiDivider-root MuiDivider-light"
      />
      <div
        class="styles__ExpansionPanelStyle-qa8k4k-9 iyqZhl"
      >
        <hr
          class="MuiDivider-root MuiDivider-light"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addBtn-275 MuiButton-textPrimary makeStyles-textPrimary-500"
            style="margin-right: auto; pointer-events: none; opacity: 0.5;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-label-499"
            >
              <span>
                Add Recipient
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-column2-288 MuiGrid-item"
  >
    <div
      class="MuiPaper-root makeStyles-paper-315 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-widgetHeader-326 makeStyles-widgetPadding-327 MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Snapshot
      </p>
      <hr
        class="MuiDivider-root MuiDivider-light"
      />
      <div
        class="styles__ExpansionPanelStyle-qa8k4k-9 iyqZhl"
      >
        <hr
          class="MuiDivider-root MuiDivider-light"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addBtn-275 MuiButton-textPrimary makeStyles-textPrimary-534"
            style="margin-right: auto; pointer-events: initial;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-label-533"
            >
              <span>
                Full Profile
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-activityContainer-274 MuiGrid-item"
  />
</div>
`;
